### PR TITLE
D8CORE-2764: menu underline fix

### PR DESCRIPTION
**Pull request by cjwest**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- Fix to the navigation menu styles

# Needed By (Date)
- Soon

# Urgency
- for v1.5.1 release

# Steps to Test

1. Check out this branch,
2. Clear cache
3. Verify that the underlines on navigation only display on hover

Screen Shot 2020-11-30 at 8 46 36 PM

Default
Screen Shot 2020-11-30 at 8 51 56 PM

Hover over external link
Screen Shot 2020-12-02 at 12 24 19 PM

# Affected Projects or Products
D8Core

# Associated Issues and/or People
* JIRA ticket: 
** https://stanfordits.atlassian.net/browse/D8CORE-2764
** https://stanfordits.atlassian.net/browse/D8CORE-3147
** https://stanfordits.atlassian.net/browse/D8CORE-3146

- Other PRs: https://github.com/SU-SWS/stanford_profile_helper/pull/51
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)

I've put all the styles for the external links and menu links in the one general file. I'm wondering if we should break them apart?

Also, Ishi is considering that she'd like to see at least some of these styles go into Decanter. 

- Anyone who should be notified? (`@mention` them here)
@ishahism 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss
Comment (sherakama) [inline modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:193]: This selector seems a bit aggressive as it applies to all navs everywhere and I could see a behavior that is different in the future. Could this selector be narrowed to the specific navigation elements you want it for?

File: modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss
Comment (cjwest) [inline modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: @sherakama,
Thanks for your help on this. This is more specific, but kinda ugly. I'm wondering if you have any suggestions?

Comment (sherakama) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: Does `.su-local-footer` not work?

Comment (sherakama) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: or `.su-local-footer__columns`

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: Yes, @sherakama, they will work. 
I was trying to find a way to only target the menus, which, clearly I'm not 😭 
If someone were to put another block into this space, I didn't want to target any links in anything but the menu blocks. Is there a rule like:

[class^="su-local-footer__" && class$="links"]
Since I want to target :
`su-local-footer__primary-links`
`su-local-footer__secondary-links`
... and anything else with `su-local-footer__*-links`

Comment (jenbreese) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: What about nav[class$="-links"] ?

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:190]: Thank you @jenbreese. That helped me think through this!

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:190]: @jenbreese Ready for your review!

Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:190]: @shea
In this ticket: https://stanfordits.atlassian.net/browse/D8CORE-3146
Ishi is changing the focus and hover state on the secondary nav from red to black
I see that these colors are set here:
https://github.com/SU-SWS/decanter/blob/master/core/src/scss/utilities/variables/components/_secondary-nav.scss#L97
Should these color change go back to Decanter? Or should they stay here?

@jenbreese FYI

Comment (sherakama) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:190]: I will defer to @kerri-augenstein for visual changes such as color. I would like them to be in the Decanter main project if we can, but it will need @kerri-augenstein's blessing.

Comment (ishahism) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:190]: I spoke with @kerri-augenstein and @yvonnetangsu regarding the color changes. They are on board with the changes we’ve noted.

File: modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss
Comment (cjwest) [inline modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:55]: this is already specified in base.css

File: modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss
Comment (cjwest) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: @jenbreese,
this is ready for review. 
Most of this PR targets the external link styles which cannot be moved to decanter because they rely on the external link module. 
Anything else that can be moved to Decanter will be done in a separate ticket and PR. 

See https://stanfordits.atlassian.net/browse/D8CORE-3170

**Review by jenbreese — APPROVED** [review]
This is working for me. I tried on both my cardinalsites and soe.

File: modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss
Comment (jenbreese) [reply modules/stanford_profile_styles/lib/scss/stanford_profile_styles.scss:189]: worked fine for me. for SOE and Core

**Comment by cjwest** [comment]
Thank you, @jenbreese !

**Pull request merged**
